File: ee/models/organization.py
from sqlalchemy import Column, BigInteger, String, Text, Enum, ForeignKey, UniqueConstraint
from sqlalchemy.dialects.postgresql import UUID, JSONB
import uuid
import time

from ee.database.base import EEBase
from core.models.enums import OrganizationStatus


class Organization(EEBase):
    __tablename__ = 'organizations'
    __table_args__ = (
        UniqueConstraint('slug', name='org_slug_unique'),
    )

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name = Column(String, nullable=False)
    slug = Column(String, nullable=False, index=True)
    description = Column(Text)
    logo_url = Column(String)
    website_url = Column(String)
    settings = Column(JSONB, default={})
    status = Column(Enum(OrganizationStatus), default=OrganizationStatus.ACTIVE)
    created_by = Column(BigInteger, ForeignKey('users.id'), nullable=False)
    subscription_plan = Column(String, default='free')
    subscription_status = Column(String, default='active')
    created_at = Column(BigInteger, nullable=False, default=lambda: int(time.time()))
    updated_at = Column(BigInteger, nullable=False, default=lambda: int(time.time()))

File: frontend/src/types/organization.ts
export interface OrganizationListItem {
  id: string;
  name: string;
  slug: string;
  role: string;
  joined_at: number;
}

export interface OrganizationDetails {
  id: string;
  name: string;
  slug: string;
  description: string | null;
  logo_url: string | null;
  website_url: string | null;
  status: string;
  subscription_plan: string;
  member_count: number;
  created_at: number;
  updated_at: number;
  /** Org default scheduling timezone (IANA), from the settings JSONB. */
  scheduling_timezone?: string | null;
}

export interface OrganizationUpdatePayload {
  name?: string;
  description?: string;
  logo_url?: string;
  website_url?: string;
  scheduling_timezone?: string;
}

export interface OrganizationUpdateResponse {
  id: string;
  name: string;
  slug: string;
  description: string | null;
  logo_url: string | null;
  website_url: string | null;
  message: string;
}

export interface OrganizationCreateResponse {
  id: string;
  name: string;
  slug: string;
  role: string;
}
